server/consumer.go

```diff
@@ -1677,35 +1677,22 @@ func (o *consumer) processNextMsgReq(_ *subscription, c *client, _, reply string
 	// In case we have to queue up this request. This is all on stack pre-allocated.
 	wr := waitingRequest{client: c, reply: reply, n: batchSize, noWait: noWait, expires: expires}
 
-	// If we are in replay mode, defer to processReplay for delivery.
-	if o.replay {
-		o.waiting.add(&wr)
+	if wr.noWait {
+		if o.maxp > 0 && len(o.pending) >= o.maxp {
+			sendErr(409, "Exceeded MaxAckPending")
```

Comment:
Not clear to me that in this error situation, we still continue. Is that expected?

Reply:
Yes, fixed.